Scripts/compile_core.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -e # exit on first error, and print commands with expansions
+
+if [ -z "$PLATFORM" ] || [ -z "$OS" ] || [ -z "$SDK" ]; then
+	echo "PLATFORM or OS not set."
+	exit 1
+fi
+
+if [ -z "$NAME" ] && ! [ "$PLATFORM" = "macOS" ]; then
+	echo "Empty NAME only allowed on macOS."
+	exit 1
+fi
+
+build_workspace() {
+	echo "Building $3"
+
+	xcodebuild -verbose "$1" build                               \
+	 	-workspace "$2"                                          \
+	 	-scheme "$3"                                             \
+	 	-sdk "$SDK"                                              \
+	 	-destination platform="$PLATFORM",OS="$OS",name="$NAME"  \
+	| bundle exec xcpretty -c
+}
+
+build_project() {
+	echo "Building $3"
+
+	xcodebuild -verbose "$1" build                               \
+	 	-project "$2"                                            \
+	 	-scheme "$3"                                             \
+	 	-sdk "$SDK"                                              \
+	 	-destination platform="$PLATFORM",OS="$OS",name="$NAME"  \
+	| bundle exec xcpretty -c
+}
+
+build_workspace "clean" "Framework/Lumberjack.xcworkspace" "CocoaLumberjack-Static"
+build_workspace "clean" "Framework/Lumberjack.xcworkspace" "CocoaLumberjack"
+build_workspace "clean" "Framework/Lumberjack.xcworkspace" "CocoaLumberjackSwift"
+
+build_project "" "Integration/Integration.xcodeproj" "iOSStaticLibraryIntegration"
+build_project "" "Integration/Integration.xcodeproj" "iOSFrameworkIntegration"
+build_project "" "Integration/Integration.xcodeproj" "iOSSwiftIntegration"
+
+OS="$$DEFAULT_WATCH_OS"
+SDK="watchSimulator$OS"
+build_project "" "Integration/Integration.xcodeproj" "watchOSSwiftIntegration"
```

Comment:
`echo "Empty NAME only allowed on macOS."`
But here you don't set NAME for `watchOS` simulator.

Reply:
The watch os platform runs on an iOS simulator, and `$NAME` is already set by the build matrix to "iPhone 8".

Reply:
@hhanesand 
And that is why I don't like environment variables :(